tpm/tpm.go

```diff
@@ -904,6 +904,61 @@ func ReadEKCert(rw io.ReadWriter, ownAuth digest) ([]byte, error) {
 	return ekbuf, nil
 }
 
+// NVDefineSpace defines space in NVRAM for the indicated index.
+// Definition include the access requirements for writing and reading the area
+func NVDefineSpace(rw io.ReadWriter, index tpmutil.Handle, len uint32, perms []uint32, ownAuth digest) ([]byte, error) {
+	sharedSecretOwn, osaprOwn, err := newOSAPSession(rw, etOwner, khOwner, ownAuth[:])
+	if err != nil {
+		return nil, fmt.Errorf("failed to start new auth session: %v", err)
+	}
+	defer osaprOwn.Close(rw)
+	defer zeroBytes(sharedSecretOwn[:])
+	perm := uint32(0)
+	for _, i := range perms {
+		perm += i
+	}
+	nvdata := nvDataPublic{0x0018, index, pcrInfoShort{}, pcrInfoShort{}, nvAttribute{tpmutil.Tag(0x0017), uint32(perm)}, false, false, false, len}
+	authIn := []interface{}{ordNVDefineSpace, nvdata}
+	ca, err := newCommandAuth(osaprOwn.AuthHandle, osaprOwn.NonceEven, sharedSecretOwn[:], authIn)
+	if err != nil {
+		return nil, fmt.Errorf("failed to construct owner auth fields: %v", err)
+	}
+	data, ra, ret, err := nvDefineSpace(rw, nvdata, ca)
+	if err != nil {
+		return nil, fmt.Errorf("failed to define space in NVRAM: %v", err)
+	}
+	raIn := []interface{}{ret, ordNVDefineSpace, tpmutil.U32Bytes(data)}
+	if err := ra.verify(ca.NonceOdd, sharedSecretOwn[:], raIn); err != nil {
+		return nil, fmt.Errorf("failed to verify authenticity of response: %v", err)
+	}
+	return data, nil
+}
+
+// NVWriteValue writes a given value with its length to a given adress in NVRAM
```

Comment:
spelling: "adress" -> "address".

Also, consider spelling out more explicitly what "address" means in this context. I think it means that data will be written to len(data) bytes starting at index + offset.